### ISSUE TITLE
Resampler fails for pd.Series if the column is not selected

### ISSUE THREAD

**Issue by Alexander-Serov** [predict-idlab/plotly-resampler#31]
Normally, plotly allows one not to specify (the only) column of a series one is plotting. So the following code works fine:
```
fig = go.Figure()
ser = pd.Series(index = np.arange(100), data = {'a': np.arange(100)})
fig.add_trace(
go.Scattergl(
    x=ser.index,
    y=ser,
)
fig.show()
```

However, it fails when combining it with the `FigureResampler` with the following error:
```
C:\tools\miniconda3\envs\stat-analysis-3.7\lib\site-packages\plotly_resampler\figure_resampler.py in add_trace(self, trace, max_n_samples, downsampler, limit_to_view, hf_x, hf_y, hf_hovertext, **trace_kwargs)
    531             if pd.isna(hf_y).any():
    532                 not_nan_mask = ~pd.isna(hf_y)
--> 533                 hf_x = hf_x[not_nan_mask]
    534                 hf_y = hf_y[not_nan_mask]
    535                 if isinstance(hf_hovertext, np.ndarray):

IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
```

The problem is solved by specifying the column of the series to plot, i.e. `y=ser['a']`. It took me several minutes to find it though and I guess `plotly-resampler` was designed to be transparent. 

Perhaps, an `if instance(pd.Series)` could be added to treat this specific case? Or could there be another way for checking if `isna` that works for such series as well?

In any way, thanks for the great software!

**Comment by jvdd** [comment]
Hi,

Thanks for creating this issue!

While trying to reproduce your error, I discovered the following;

```py 
ser = pd.Series(index=np.arange(100), data={'a': np.arange(100)})
``` 
creates a `pd.Series` with all (i.e., 100) nan-values.

Changing this to
```py
ser = pd.Series(index=np.arange(100), data=np.arange(100), name="a")
```
results in the expected outcome; a series of 100 ints.

Then, your updated snippet looks like this
```py
fig = FigureResampler(go.Figure())
ser = pd.Series(index=np.arange(100), data=np.arange(100), name="a")
ser.iloc[42] = np.nan  # set 1 value to nan
fig.add_trace(
    go.Scattergl(
        x=ser.index,
        y=ser,
    )
)
fig.show_dash()
```

Which works perfectly fine!
But when I convert `ser` to a `pd.DataFrame` (that contains a nan value), then I do get the exact same error. See snippet below.

```py
fig = FigureResampler(go.Figure())
ser = pd.Series(index=np.arange(100), data=np.arange(100), name="a").to_frame()
ser.iloc[42] = np.nan  # set 1 value to nan
fig.add_trace(
    go.Scattergl(
        x=ser.index,
        y=ser,
    )
)
fig.show_dash()
```

---

So in conclusion, this error occurs when you pass a `pd.DataFrame` (with nan values) as your `y` variable. The code seems to work as intended when a `pd.Series` (with nan values) is passed as `y` variable.

When a dataframe without nan values is passed, the behavior is exactly the same as plotly. When the dataframe contains nans, the behavior is differently (i.e., an error is thrown). In my opinion, a resolution for this issue might be implementing a check that raises a more clear error when either `x` or `y` are `pd.DataFrame`.

We are glad to hear that you like our library :)

**Comment by Alexander-Serov** [comment]
Hey @jvdd, 

Oops, yes, sorry, I have adapted the snippet from my code but had limited time, so I didn't actually launch it. Sorry if it took you more time to reproduce than it could have had. :(

And indeed, you have identified the problem correctly, my bad. I have gone back to the state of code I had and apparently I have left a single column in a data frame, but it was still a data frame. And, as you guessed, that resulted in an error that I have posted. Now I see clearly where I went the wrong way.

So since the problem was clearly on my side, I would say there is nothing to do in `plotly-resampler`. Unless you would like to add a clearer error, as you say, for this part of code that is outside plotly.

Thanks again for the prompt reply!

**Comment by Alexander-Serov** [comment]
I have encountered though another problem related to Gitlab pipelines when using `plotly-resampler`. I will post it in a separate issue in case it helps anyone.
